File: Udemy Master Data Structure and Algorithm/6 - Array/merge-sorted-array.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> mergeTwoArray(vector<int> arr1, vector<int> arr2)
{
  vector<int> mergeArray;
  // check if the input is 0 or not
  if (arr1.size() == 0)
  {
    return arr2;
  }

  if (arr2.size() == 0)
  {
    return arr1;
  }

  if (arr1.size() > arr2.size())
  {
    mergeArray = arr1;
    for (int i = 0; i < arr2.size(); i++)
    {
      mergeArray.push_back(arr2[i]);
    }
  }

  if (arr2.size() > arr1.size())
  {
    mergeArray = arr2;
    for (int i = 0; i < arr1.size(); i++)
    {
      mergeArray.push_back(arr1[i]);
    }
  }

  sort(mergeArray.begin(), mergeArray[i].end());

  return mergeArray;
}

int main()
{
  vector<int> array1 = {1, 2, 39, 239, 21, 3};
  vector<int> array2 = {78, 23, 45, 24, 58};

  vector<int> answerArray;

  answerArray = mergeTwoArray(array1, array2);

  for (int i = 0; i < answerArray.size(); i++)
  {
    cout << answerArray[i] << "," << endl;
  }
  return 0;
}

File: Udemy-Master-Data-Structure-and-Algorithm/6 - Array/merge-sorted-array.js
// Merged two array and sort it

const array1 = [1, 2, 39, 239, 21, 3];
const array2 = [78, 23, 45, 24, 58];

function mergeTwoArray(arr1, arr2) {
  const mergedArray = arr1;

  // check the input
  if (arr1.length === 0) {
    return arr2;
  }

  if (arr2.length === 0) {
    return arr1;
  }

  if (arr1.length > arr2.length) {
    const lastItemsArr = arr2.length - 1;
    for (let i = lastItemsArr; i >= 0; i--) {
      mergedArray.push(arr2[i]);
    }
  }

  if (arr2.length > arr1.length) {
    const lastItemsArr = arr1.length - 1;
    for (let i = lastItemsArr; i >= 0; i--) {
      mergedArray.push(arr1[i]);
    }
  }

  mergedArray.sort((a, b) => a - b);
  return mergedArray;
}

console.log(mergeTwoArray(array1, array2));
